Sources/SwiftProtobuf/_MessageStorage.swift

```diff
@@ -718,6 +746,25 @@ extension _MessageStorage {
             updateValue(at: offset, to: newValue, oneofPresence: (oneofOffset, field.fieldNumber))
         }
     }
+
+    /// Appends the given value to the values already present in the field, initializing the field
+    /// if necessary.
+    func appendValue<T>(_ value: T, to field: FieldLayout) {
+        // If the field isn't already present, we need to initialize a new array first.
+        let pointer = (buffer.baseAddress! + field.offset).bindMemory(to: [T].self, capacity: 1)
+        if !isPresent(field) {
```

Comment:
To append, doesn't it already have to have an array in there?  i.e. - you create the array above in a bunch of cases.  i.e. - when would it need to actually have to do this initialize and set of the has bit vs. be able to assume it is already done?

Reply:
Decoding any value in a repeated field is an "append" operation since those are the merge semantics. Let's say you have a message with a repeated message field, and you create a new instance of it. Then, you decode/merge into it. When you find that message field on the wire, you decode it and "append" it to what's already stored there (which might be nothing). For the first one we encounter that means the array hasn't actually been initialized yet. So we have to handle that case.

Reply:
Chatted offline, figured out where I was misconnecting things.